### ISSUE TITLE
UnicodeDecodeError while testing NX OS derived Yang model

### ISSUE THREAD

**Issue by ygorelik** [CiscoDevNet/ydk-gen#674]
Issue tracker is ONLY used for reporting bugs. Please use the [YDK Community](https://communities.cisco.com/community/developer/ydk) for any support issues.

## Current Behavior
While running codec tests of NX OS derived model ydktest-sanity-typedefs.yang in Python we get error originated in pybind11:
UnicodeDecodeError: 'utf8' codec can't decode byte 0xc0 in position 31: invalid start byte

## Steps to Reproduce

Add the following test to the test_sanity_codec.py and run the test.
```python
   def test_encode_decode_typedefs(self):
        from ydk.models.ydktest import ydktest_sanity_typedefs as ysanity_typedefs
        system_encode = ysanity_typedefs.System()
        system_encode.mode = ysanity_typedefs.TopMode.stand_alone
        system_encode.id = 22;
        
        #self.provider.encoding = EncodingFormat.XML
        payload = self.codec.encode(self.provider, system_encode)
        
        system_decode = self.codec.decode(self.provider, payload)
        self.assertEqual(system_encode, system_decode)
```
## Logs

```
warning: Debugger speedups using cython not found. Run '"/usr/bin/python" "/home/yan/.p2/pool/plugins/org.python.pydev_6.2.0.201711281614/pysrc/setup_cython.py" build_ext --inplace' to build.
pydev debugger: starting (pid: 23670)
test_encode_decode_typedefs (__main__.SanityYang) ... ERROR

======================================================================
ERROR: test_encode_decode_typedefs (__main__.SanityYang)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yan/ydk-workspace/ydk-gen/sdk/python/core/tests/test_sanity_codec.py", line 367, in test_encode_decode_typedefs
    payload = self.codec.encode(self.provider, system_encode)
  File "/usr/local/lib/python2.7/dist-packages/ydk/errors/error_handler.py", line 112, in helper
    return func(self, provider, entity, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/ydk/services/codec_service.py", line 74, in encode
    return self._encode(provider, entity_holder, pretty, subtree)
  File "/usr/local/lib/python2.7/dist-packages/ydk/services/codec_service.py", line 101, in _encode
    data_node = _get_data_node_from_entity(entity, root_schema)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xc0 in position 31: invalid start byte

----------------------------------------------------------------------
Ran 1 test in 97.052s

FAILED (errors=1)
```

## System Information

Tested in Ubuntu 14.1

**Comment by ygorelik** [comment]
The issue is caused by the following statement in the NX OS Yang model:

    // Dot 1Q Ethertype
    typedef l1_Dot1qEtherType {
        type enumeration {
            enum 0x8100 {
                value 1;
            }
            enum 0x88A8 {
                value 2;
            }
            enum 0x9100 {
                value 3;
            }
            enum any {
                value 4;
            }
        }
        default "0x8100";
    }

Note that enum key appears as hexadecimal notation, which confuses the Python11.

**Comment by ygorelik** [comment]
The issue is originating in the libyang library. See details [here](https://github.com/CESNET/libyang/issues/446).

**Comment by ygorelik** [comment]
Tested YDK after applying libyang fix (merged with version 0.14.70). All worked as expected. 

See logs and test script for details: 
[test_674.zip](https://github.com/CiscoDevNet/ydk-gen/files/1752652/test_674.zip)

**Comment by abhikeshav** [comment]
Thanks @ygorelik !
